extensions/tika/runtime/src/main/java/io/quarkus/tika/TikaParser.java

```diff
@@ -68,8 +87,11 @@ protected TikaContent parseStream(InputStream entityStream, String contentType,
     }
 
     private Parser createParser() {
-        // TODO Use the properties to support loading the format-specific and other generic parsers
-        return new AutoDetectParser();
+        return new AutoDetectParser(tikaConfig);
+    }
+
+    private ContentHandler createContentHandler() {
+        return new ToTextContentHandler();
```

Comment:
So what you could do here is make the default content handler a bean marked with `@DefaultBean`. Then the user would be able to easily tweak the default one.

Reply:
That one, as well as other often used ones are inside the Tika `tiika-parsers` module, this is why I thought adding few overrides to `TikaParser` can close the gap, and ContentHandler has to be instantiated per request, so I'm not sure yet if `@DefaultBean` can help. 
The other point is that the content per embedded document approach complicates things further, so I'm adding some content handler validation and some specific instantiation if that strategy is chosen. Hoping to update the PR shortly (with the above 2 comments being addressed) and then I'll ping you to do the review :-)